e2e/ansible/playbooks/hyperconverged/test-k8s-jenkins-upgrade/k8s-jenkins-upgrade.yml

```diff
@@ -0,0 +1,232 @@
+
+###############################################################################################
+#Test Steps:
+#1. Check whether the OpenEBS components are deployed.
+#2. Download the test artifacts to k8s master.
+#3. Deploy Jenkins application.
+#4. Check if the application pod is up and running
+#5. Check if the Jenkins service us up.
+#6. Perform Job creation and building activity inside pod.
+#7. Check for data persitency.
+###############################################################################################
+
+- hosts: localhost
+
+  vars_files:
+    - k8s-jenkins-pod-vars.yml
+ 
+  tasks:
+
+   - block:
+
+     - include: k8s-jenkins-pod-prereq.yml
+      
+     - name: Download YAML for jenkins 
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/get_url.yml"
+       vars:
+         url: "{{ jenkins_link }}"
+         dest: "{{ result_kube_home.stdout }}/{{ pod_yaml_alias }}"
+
+     - name: Check whether maya-apiserver is deployed
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+       vars:
+          ns: "{{ operator_ns }}"
+          app: maya-api
+
+     - name: Replace jenkins pod version
+       lineinfile:
+         path: "{{ result_kube_home.stdout }}/{{ pod_yaml_alias }}"
+         regexp: "image: jenkins/jenkins:"
+         line: "     image: jenkins/jenkins:{{jenkinsold}}"
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+    
+     - name: Create namespace for deployement.
+       shell: source ~/.profile; kubectl create ns {{namespace}}
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+      
+     - name:  Deploy jenkins pod 
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_task.yml"
+       vars:
+          app_yml: "{{ pod_yaml_alias }}"
+          ns: "{{ namespace }}"
+    
+     - name:  Confirm pod status is running
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+       vars:
+          ns: "{{ namespace }}"
+          app: jenkins
+
+     - name: Verify that the jenkins is a cluster service
+       shell: source ~/.profile; kubectl get svc -n {{namespace}}
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: result_service
+       failed_when: "'jenkins-svc' not in result_service.stdout"
+
+     - name: Get the name of the pod.
+       shell: source ~/.profile; kubectl get pods -n {{namespace}} | grep jenkins | awk {'print $1'}
+       args: 
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: jenkins_pod_name
+
+         
+     - name: Download Jenkins-cli.jar file inside /var/jenkins_home of pod.
+       shell: >
+          source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }} 
+          -- wget -P /var/jenkins_home http://localhost:8080/jnlpJars/jenkins-cli.jar
```

Comment:
Can you explain this download step from localhost ?

Reply:
its downloading and saving jenkins-cli.jar file inside mounted path of jenkins app pod (var/jenkins_home) directly.